core/assay.js

```diff
@@ -0,0 +1,133 @@
+      // an amount so labeling that quantity.
+      make(allegedQuantity) {
+        const amount = harden({
+          label,
+          quantity: strategy.insistKind(allegedQuantity),
+        });
+        brand.add(amount);
+        return amount;
+      },
+
+      // Is this an amount object made by this assay? If so, return
+      // it. Otherwise error.
+      vouch(amount) {
+        insist(brand.has(amount))`\
+  Unrecognized amount: ${amount}`;
+        return amount;
+      },
+
+      // Is this like an amount object made by this assay, such as one
+      // received by pass-by-copy from an otherwise-identical remote
+      // amount? On success, return an amount object made by this
+      // assay. Otherwise error.
+      //
+      coerce(allegedAmount) {
+        if (brand.has(allegedAmount)) {
+          return allegedAmount;
+        }
+        if (!Object.prototype.hasOwnProperty.call(allegedAmount, 'quantity')) {
+          // this is not an amount, let's see if it's a quantity
+          // Will throw on inappropriate quantity
+          return assay.make(allegedAmount);
+        }
+        const { label: allegedLabel, quantity } = allegedAmount;
+        mustBeSameStructure(label, allegedLabel, 'Unrecognized label');
+        // Will throw on inappropriate quantity
+        return assay.make(quantity);
+      },
+
+      // Return the raw quantity that this amount labels.
+      quantity(amount) {
+        return assay.vouch(amount).quantity;
+      },
+
+      // Represents the empty set of erights, i.e., no erights
+      empty() {
+        return assay.make(strategy.empty());
+      },
+
+      isEmpty(amount) {
+        return strategy.isEmpty(assay.quantity(amount));
+      },
+
+      // Set inclusion of erights.
+      // Does the set of erights described by `leftAmount` include all
+      // the erights described by `rightAmount`?
+      includes(leftAmount, rightAmount) {
+        const leftQuantity = assay.quantity(leftAmount);
+        const rightQuantity = assay.quantity(rightAmount);
+        return strategy.includes(leftQuantity, rightQuantity);
+      },
+
+      equals(leftAmount, rightAmount) {
+        const leftQuantity = assay.quantity(leftAmount);
+        const rightQuantity = assay.quantity(rightAmount);
+        return strategy.equals(leftQuantity, rightQuantity);
+      },
+
+      // Set union of erights.
+      // Describe all the erights described by `leftAmount` and those
+      // described by `rightAmount`.
+      with(leftAmount, rightAmount) {
+        const leftQuantity = assay.quantity(leftAmount);
+        const rightQuantity = assay.quantity(rightAmount);
+        return assay.make(strategy.with(leftQuantity, rightQuantity));
+      },
+
+      // Covering set subtraction of erights.
+      // If leftAmount does not include rightAmount, error.
+      // Describe the erights described by `leftAmount` and not described
```

Comment:
The repetition of Describe/described doesn't help. How about something like "Return a representation of those rights included in 'leftAmount' but not included in 'rightAmount'."?